contracts/test/StakeLocker.t.sol

```diff
@@ -2,663 +2,254 @@
-        assertEq(stakeLocker1.stakeDate(address(sam)), start);         // Stays the same
-        assertEq(stakeLocker1.stakeDate(address(sid)), newStakeDate);  // Gets updated
+        assertTrue(staker.try_stakeLocker_cancelUnstake(address(locker)));
+        
+        assertEq(locker.unstakeCooldown(address(staker)), 0);
     }
 
-    function setUpLoanAndRepay() public {
-        mint("USDC", address(leo), 10_000_000 * USD);  // Mint USDC to LP
-        leo.approve(USDC, address(pool1), MAX_UINT);   // LP approves USDC
+    function test_unstake(uint256 stakeAmount, uint256 unstakeAmount) external {
+        stakeAmount = constrictToRange(stakeAmount, 1, type(uint256).max / block.timestamp);
+        unstakeAmount = constrictToRange(unstakeAmount, 1, stakeAmount);
 
-        leo.deposit(address(pool1), 10_000_000 * USD);                                        // LP deposits 10m USDC to Pool
-        pat.fundLoan(address(pool1), address(loan1), address(dlFactory1), 10_000_000 * USD);  // PD funds loan for 10m USDC
+        stakeToken.mint(address(staker), stakeAmount);
+        staker.erc20_approve(address(stakeToken), address(locker), stakeAmount);
+        delegate.stakeLocker_openStakeLockerToPublic(address(locker));
+        staker.stakeLocker_stake(address(locker), stakeAmount);
+        staker.stakeLocker_intendToUnstake(address(locker));
 
-        uint256 cReq = loan1.collateralRequiredForDrawdown(10_000_000 * USD);  // WETH required for 100_000_000 USDC drawdown on loan
-        mint("WETH", address(bob), cReq);                                      // Mint WETH to borrower
-        bob.approve(WETH, address(loan1), MAX_UINT);                           // Borrower approves WETH
-        bob.drawdown(address(loan1), 10_000_000 * USD);                        // Borrower draws down 10m USDC
+        uint256 start = block.timestamp;
 
-        mint("USDC", address(bob), 10_000_000 * USD);  // Mint USDC to Borrower for repayment plus interest
-        bob.approve(USDC, address(loan1), MAX_UINT);   // Borrower approves USDC
-        bob.makeFullPayment(address(loan1));           // Borrower makes full payment, which includes interest
+        // Can't unstake immediately (lockup not expired)
+        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));
 
-        pat.claim(address(pool1), address(loan1), address(dlFactory1));  // PD claims interest, distributing funds to stakeLocker
-    }
+        hevm.warp(start + globals.stakerCooldownPeriod() - 1);
 
-    function test_unstake(uint256 stakeAmount) public {
-        uint256 bptMin = WAD / 10_000_000;
-        stakeAmount = constrictToRange(stakeAmount, bptMin, bPool.balanceOf(address(sam)), true);  // 25 WAD max, 1/10m WAD min, or zero (min is roughly equal to 10 cents) (non-zero)
+        // Can't unstake in cooldown (lockup not expired)
+        assertTrue(!staker.try_stakeLocker_unstake(address(locker), unstakeAmount));
```

Comment:
We can actually remove these three lines, and then make like 168
`hevm.warp(start + globals.stakerCooldownPeriod()));`

We only care that the lockupPeriod failure mode is triggered